archunit/src/main/java/com/tngtech/archunit/core/domain/Formatters.java

```diff
@@ -92,7 +92,7 @@ public static List<String> formatNamesOf(Class<?>... paramTypes) {
      * @return A {@link List} of fully qualified class names of the passed {@link Class} objects
      */
     @PublicAPI(usage = ACCESS)
-    public static List<String> formatNamesOf(Iterable<Class<?>> paramTypes) {
+    public static List<String> formatNamesOf(Iterable<? extends Class<?>> paramTypes) {
```

Comment:
That is part of the Java Generics variance behavior. At least AFAIS without the `? extends` you can't use it for

```
Formatters.formatNamesOf(ImmutableList.of(String.class, Serializable.class));
```

while you can use it for

```
Formatters.formatNamesOf(ImmutableList.of(String.class, Object.class));
```

The problem seems to be that in the former case the inferred type is `List<Class<? extends Serializable>>` which seems to conflict with `Iterable<Class<?>>`. So to make the API more convenient I changed the signature to allow subtypes of `Class<?>`, because then the former case compiles without the need to explicitly fix the type by `ImmutableList.<Class<?>>`. Should maybe add a test though and split out the commit :wink: (because I saw that the final state actually compiles without this change, so I think I added it at some point when I stumbled over it, but the reason afterwards wasn't necessary to be added to the PR after all :man_shrugging:)